### DISCUSSION TITLE
nvim-java with NvChad v2.5

### DISCUSSION THREAD

**Discussion by Jamie-505** [nvim-java/nvim-java#308] [Show and tell]
After almost pulling my hair out because nvchads mason config did not play nicely with the extensions from nvim-java I found this [reddit post](https://www.reddit.com/r/neovim/comments/1bdsr3t/how_i_got_nvimjava_to_work_fully_in_nvchad/?show=original) Credit: @Omixxx

so I figured I post this somewhere more easily to find.

here is the **lua/plugins/myPlugins.lua** section:
```lua
  {
    'nvim-java/nvim-java',
    lazy = false,
    dependencies = {
      'nvim-java/lua-async-await',
      'nvim-java/nvim-java-core',
      'nvim-java/nvim-java-test',
      'nvim-java/nvim-java-dap',
      'MunifTanjim/nui.nvim',
      'neovim/nvim-lspconfig',
      'mfussenegger/nvim-dap',
      {
        'williamboman/mason.nvim',
        opts = {
          registries = {
            'github:nvim-java/mason-registry',
            'github:mason-org/mason-registry',
          },
        },
      },
    },
    config = function()
      require('configs.java')
    end,
  },
```

and the **lua/configs/java.lua** file I use to store the configuration

```lua
local config = require('nvchad.configs.lspconfig')
local on_attach = config.on_attach
local capabilities = config.capabilities

require('java').setup({})
require('lspconfig').jdtls.setup({
  on_attach = on_attach,
  capabilities = capabilities,
  filetypes = { 'java' },
})
```

hope this serves useful for someone

**Comment by CodEssence** [comment]
Hey mate!

i also came accross this issue, but i use latest version of **nvchad**. 

**here i share my setup:**

**_note:_** nvchad config structure canged since then and now custom folder is no longer available. but i like to use a structure similar to custom folder. here is my config file structure:

![filetree](https://github.com/user-attachments/assets/0a225c6c-6170-47cc-822b-a34238829422)



- **init.lua file** (central point for loading all my plugins):




```lua

-- Updated config --

-- ~/.config/nvim/lua/plugins/init.lua

-- Create an empty table to hold all plugin specifications
local plugins = {}

-- Helper function to extend the 'plugins' table with configurations from other files
local function extend_plugins(t)
  for _, v in ipairs(t) do
    table.insert(plugins, v)
  end
end

--- Load existing plugins from your original init.lua directly into the 'plugins' table ---

-- These are the plugins you had directly listed in your original init.lua.
-- Add them to the 'plugins' table here.
table.insert(plugins, {
  "stevearc/conform.nvim",
  -- event = 'BufWritePre', -- uncomment for format on save
  opts = require "configs.conform",
})

table.insert(plugins, {
  "neovim/nvim-lspconfig",
  config = function()
    require "configs.lspconfig"
  end,
})

-- Uncomment and add other plugins you had directly here, for example:
-- table.insert(plugins, { import = "nvchad.blink.lazyspec" })

-- table.insert(plugins, {
--   "nvim-treesitter/nvim-treesitter",
--   opts = {
--     ensure_installed = {
--       "vim", "lua", "vimdoc",
--       "html", "css"
--     },
--     -- Ensure folding is enabled if you plan to use Treesitter folding
--     folding = { enable = true },
--   },
-- })

-- == uncommented: ==

 table.insert(plugins, { import = "nvchad.blink.lazyspec" })

 table.insert(plugins, {
   "nvim-treesitter/nvim-treesitter",
   opts = {
     ensure_installed = {
       "vim", "lua", "vimdoc",
       "html", "css", "java", "python"
     },
     -- Ensure folding is enabled if you plan to use Treesitter folding
     folding = { enable = true },
   },
 })


-- Add Mason and Mason-LSPConfig here
table.insert(plugins, {
  "williamboman/mason.nvim",
  cmd = "Mason",
  build = ":MasonUpdate",
  config = function()
    require("mason").setup({
      -- Add any global mason settings here if needed
      -- e.g., ensure that the "bin" directory is in your system PATH
    })
  end
})

table.insert(plugins, {
  "williamboman/mason-lspconfig.nvim",
  dependencies = { "williamboman/mason.nvim", "neovim/nvim-lspconfig" },
  config = function()
    require("mason-lspconfig").setup({
      ensure_installed = {
        "jdtls",           -- Ensure JDTLS is always installed by Mason
        "lua_ls",          -- Ensure Lua LS is installed
        "cssls",           -- Ensure CSS LS is installed
        "html"             -- Ensure HTML LS is installed
      },
      -- Automatically setup lspconfig for servers installed by mason
      automatic_installation = true,
    })
  end
})


--- Load plugins from different groups (your new structure) ---

-- Load plugins from the 'note-taking' group
extend_plugins(require("plugins.note-taking.telekasten"))
extend_plugins(require("plugins.note-taking.render-markdown"))

-- Load plugins from the 'java' development group
extend_plugins(require("plugins.java.nvim-java"))

-- Add more groups as you create them, for example:
-- extend_plugins(require("plugins.utility.harpoon"))
-- extend_plugins(require("plugins.lsp.nvim-lspconfig"))

-- Finally, return the combined list of all plugin specifications for lazy.nvim to process
return plugins



```


- then i have **nvim-java.lua** file:


![nvimjava](https://github.com/user-attachments/assets/b4be4946-4909-4439-9fe1-14dc27be333d)



- also the file i store the **configs** _like you do:_
(i used to have both in the same file but i saw your setup and wanted to try it out)


![configjava](https://github.com/user-attachments/assets/1baa4a0b-edd9-44af-bb0e-e918b1112088)




Long story short i wanted to learn java and got nvchad, tweaked it for a week or so now i am trying to make java working. but after long hours of trying to set this up in nvchad it wont still work. i got the configurations from that archved post on **reddit**, then i saw your post, and tried that, but it still is not working:

- Lazy installs the packages needed when i first open it up, but..
- commands aint working and those wont pop up on autocomplete;
- treesitter appears to be partially working (e.g., syntax highlighting seems okay), but I'm unsure if it's fully integrated for Java features as I'm new to it.
- the debugger does not appear when I attempt to run the commands shown in nvim-java demos.
etc.

could you help over please! 🙏

**Reply by CodEssence** [reply]
**Update:** i added my fdk path to lua/config/nvim-java.lua file i had. **But it still doesnt work :(**
now it looks like this:
```lua
-- File: ~/.config/nvim/lua/configs/nvim-java.lua

local config = require('nvchad.configs.lspconfig')
local on_attach = config.on_attach
local capabilities = config.capabilities

require('java').setup({})

require('lspconfig').jdtls.setup({
    settings = {
        java = {
            configuration = {
                runtimes = {
                    {
                        name = "JavaSE-22",
                        path = "/usr/lib/jvm/jdk-22-oracle-x64",
                        default = true,
                    },
                },
            },
        },
    },
    on_attach = function(client, bufnr)
        require('java').dap.on_attach(client, bufnr)
        on_attach(client, bufnr)
    end,
    capabilities = capabilities,
    filetypes = { 'java' },
})
```

**But it still doesnt work :(**
